sylt/src/formatter.rs

```diff
@@ -0,0 +1,504 @@
+use std::fmt::{self, Write};
+use std::path::PathBuf;
+use sylt_common::{Error, Type as RuntimeType};
+use sylt_parser::statement::UseIdentifier;
+use sylt_parser::{
+    Assignable, AssignableKind, Expression, ExpressionKind, Identifier, Module, Op, Statement,
+    StatementKind, Type, TypeKind, VarKind,
+};
+
+use crate::Args;
+
+static INDENT: &'static str = "    ";
+
+macro_rules! write_comma_separated {
+    ($dest:expr, $indent:expr, $write:expr, $values:expr) => {
+        let mut first = true;
+        for value in $values {
+            if !first {
+                write!($dest, ", ")?;
+            }
+            first = false;
```

Comment:
This could be an enum? So we don't have to mutate stuff?

Reply:
Why would that not need to mutate?